core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -96,45 +114,76 @@ oppia.controller('FeedbackTab', [
       backdrop: true,
       size: 'lg',
       resolve: {
-        suggestion: function() {
-          return $scope.activeThread.suggestion;
+        suggestionInfo: function() {
+          return {
+            isSuggestionValid: $scope.isSuggestionValid(),
+            hasUnsavedChanges: $scope.hasUnsavedChanges()
+          };
+        },
+        stateContent: function() {
+          var states = explorationData.data.states;
+          var stateName = $scope.activeThread.suggestion.state_name;
+          return {
+            oldContent: states[stateName].content[0].value,
+            newContent: $scope.activeThread.suggestion.state_content.value
+          };
         }
       },
       controller: [
-        '$scope', '$modalInstance', 'suggestion',
-        function($scope, $modalInstance, suggestion) {
-        $scope.oldContent = suggestion.state_content.old_content;
-        $scope.newContent = suggestion.state_content.new_content;
+        '$scope', '$modalInstance', 'suggestionInfo', 'stateContent',
+        function($scope, $modalInstance, suggestionInfo, stateContent) {
+        var ACTION_INVALID_MSG = 'This suggestion has already been ' +
+          'acted upon, or was made for an outdated version '  +
+          'of this exploration.';
+        var UNSAVED_CHANGES_MSG = 'You have unsaved changes to this ' +
+          'exploration. Please save your changes before acting on ' +
+          'a suggestion.';
+        $scope.canActOnSuggestion = (
+          suggestionInfo.isSuggestionValid && !suggestionInfo.hasUnsavedChanges
+        );
+        $scope.errorMessage = suggestionInfo.isSuggestionValid
```

Comment:
Probably more conceptually correct for this to be:

    $scope.errorMessage = (
      !suggestionInfo.isSuggestionValid ? ACTION_INVALID_MSG : 
      suggestionInfo.hasUnsavedChanges ? UNSAVED_CHANGES_MSG : 
      '');

Basically, the JS should make sense without having to know the structure of the DOM.